source/server/connection_handler_impl.cc

```diff
@@ -460,9 +460,10 @@ void ConnectionHandlerImpl::ActiveTcpListener::resumeListening() {
 
 void ConnectionHandlerImpl::ActiveTcpListener::newConnection(
     Network::ConnectionSocketPtr&& socket, std::unique_ptr<StreamInfo::StreamInfo> stream_info) {
-  // Refresh local address in case it was restored by a listener filter like the original_dst
-  // filter.
+  // Refresh addresses in case they are modified by listener filters, such as proxy protocol or
+  // original_dst.
   stream_info->setDownstreamLocalAddress(socket->localAddress());
+  stream_info->setDownstreamRemoteAddress(socket->remoteAddress());
```

Comment:
Can you cut a follow up ticket about auditing this code elsewhere? It think there may be code in the HCM we don't or shouldn't need assuming stream info chaining is working correctly?

Reply:
#14133